specifyweb/frontend/js_src/lib/components/Formatters/Formatter.tsx

```diff
@@ -169,78 +237,103 @@ function Fields({
   readonly fields: GetSet<Formatter['definition']['fields'][number]['fields']>;
   readonly onDelete: (() => void) | undefined;
 }): JSX.Element {
+  const [displayFormatter, setDisplayFormatter] = React.useState(false);
   return (
-    <table
-      /*
-       * REFACTOR: replace min-w-[35rem] with a container query that replaces
-       *   table layout with list layout
-       */
-      className={`
-        grid-table min-w-[35rem] grid-cols-[min-content_auto_auto_min-content]
-        gap-2
-      `}
-    >
-      <thead>
-        <tr>
-          <th>{resourcesText.separator()}</th>
-          <th>{schemaText.field()}</th>
-          <th>{resourcesText.formatter()}</th>
-          <td />
-        </tr>
-      </thead>
-      <tbody>
-        {fields.map((field, index) => (
-          <Field
-            field={[
-              field,
-              (field): void => setFields(replaceItem(fields, index, field)),
-            ]}
-            key={index}
-            table={table}
-            onRemove={(): void => setFields(removeItem(fields, index))}
-          />
-        ))}
-        <tr>
-          <td className="col-span-3 !gap-2">
-            <Button.Success
-              onClick={(): void =>
-                setFields([
-                  ...fields,
-                  {
-                    separator: localized(' '),
-                    aggregator: undefined,
-                    formatter: undefined,
-                    fieldFormatter: undefined,
-                    field: undefined,
-                  },
-                ])
-              }
-            >
-              {resourcesText.addField()}
-            </Button.Success>
-            {typeof handleDelete === 'function' && (
-              <Button.Danger onClick={handleDelete}>
-                {resourcesText.deleteDefinition()}
-              </Button.Danger>
-            )}
-          </td>
-          <td />
-        </tr>
-      </tbody>
-    </table>
+    <>
+      {fields.length === 0 ? null : (
+        <table
+          /*
+           * REFACTOR: replace min-w-[35rem] with a container query that replaces
+           *   table layout with list layout
```

Comment:
if you are curious about this and want to explore more, here is the tailwind plugin for container queries: https://github.com/tailwindlabs/tailwindcss-container-queries

it's an official plugin from the developers of tailwind and they talked how it would be integrated into the core tailwind in the future.

container queries are cool and useful, especially for complicated layouts like those in sp7 - they can improve adaptive design and aid in code reuse - if you learn them now, you will find how to apply them in other places
(i.e with container queries, the page could display differently based on whether the side menu is collapsed or not as that affects available space - without container queries you can only listen for browser window size change, not the container size change, which can be not enough)